subsys/emul/espi/espi_emul_host.c

```diff
@@ -0,0 +1,388 @@
+
+	return 0;
+}
+
+/* Logical Channel 2 APIs */
+static int host_espi_api_send_oob(const struct espi_emul *dev,
+				 struct espi_oob_packet *pckt)
+{
+	return -EIO;
+}
+
+static int host_espi_api_receive_oob(const struct espi_emul *dev,
+				    struct espi_oob_packet *pckt)
+{
+	return -EIO;
+}
+
+/* Logical Channel 3 APIs */
+static int host_espi_api_flash_read(const struct espi_emul *dev,
+				   struct espi_flash_packet *pckt)
+{
+	return -EIO;
+}
+
+static int host_espi_api_flash_write(const struct espi_emul *dev,
+				    struct espi_flash_packet *pckt)
+{
+	return -EIO;
+}
+
+static int host_espi_api_flash_erase(const struct espi_emul *dev,
+				    struct espi_flash_packet *pckt)
+{
+	return -EIO;
+}
+
+/* Callbacks and traffic intercept */
+static int host_espi_api_manage_callback(const struct espi_emul *dev,
+					struct espi_callback *callback,
+					bool set)
+{
+	struct espi_host_emul_data *data =
+		CONTAINER_OF(dev, struct espi_host_emul_data, emul);
+
+	return espi_manage_callback(&data->callbacks, callback, set);
+}
+
+int host_espi_api_raise_event(const struct espi_emul *dev, struct espi_event ev)
+{
+	struct espi_host_emul_data *data =
+		CONTAINER_OF(dev, struct espi_host_emul_data, emul);
+
+	espi_send_callbacks(&data->callbacks, data->espi, ev);
+
+	return 0;
+}
+
+static struct espi_emul_device_api espi_emul_host_api = {
+	.config = host_espi_api_config,
+	.get_channel_status = host_espi_api_get_channel_status,
+	.read_request = host_espi_api_read_request,
+	.write_request = host_espi_api_write_request,
+	.read_lpc_request = host_espi_api_lpc_read_request,
+	.write_lpc_request = host_espi_api_lpc_write_request,
+	.send_vwire = host_espi_api_send_vwire,
+	.receive_vwire = host_espi_api_receive_vwire,
+	.internal_set_vwire = host_espi_api_internal_set_vwire,
+	.send_oob = host_espi_api_send_oob,
+	.receive_oob = host_espi_api_receive_oob,
+	.flash_read = host_espi_api_flash_read,
+	.flash_write = host_espi_api_flash_write,
+	.flash_erase = host_espi_api_flash_erase,
+	.manage_callback = host_espi_api_manage_callback,
+	.raise_event = host_espi_api_raise_event,
+};
+
+/**
+ * Set up a new eSPI host emulator
+ *
+ * @param emul Emulation information
+ * @param bus Device to emulated eSPI controller
+ * @return 0 indicating success (always)
+ */
+static int emul_host_init(const struct emul *parent, const struct device *bus)
+{
+	const struct espi_host_emul_cfg *cfg = parent->cfg;
+	struct espi_host_emul_data *data = parent->data;
+	int a;
+
+	data->emul.api = &espi_emul_host_api;
+	data->emul.chipsel = cfg->chipsel;
+	data->emul.parent = parent;
+	data->espi = bus;
+	emul_host_init_vw_state(data);
+
+#ifdef CONFIG_ESPI_PERIPHERAL_ACPI_SHM_REGION
+	for (a = 0; a < CONFIG_ESPI_EMUL_HOST_ACPI_SHM_REGION_SIZE; a++) {
+		data->shm_acpi_mmap[a] = 0;
+	}
+#endif
+
+#ifdef CONFIG_ESPI_PERIPHERAL_CUSTOM_OPCODE
+	for (a = 0; a < CONFIG_ESPI_EMUL_HOST_CMD_SHM_REGION_SIZE; a++) {
+		data->shm_host_cmd[a] = 0;
```

Comment:
Could this just be something like `memset(data->shm_host_cmd, sizeof(data->shm_host_cmd), 0);`?

Reply:
Sure, done